File: breaking-the-records.py
def breakingRecords(scores):
    maxi = mini = scores[0]
    max_count = min_count = 0
    
    for i in range(1,len(scores)):
        a = scores[i]
        if a > maxi:
            maxi = max(maxi,a)
            max_count +=1
        elif a < mini :
            mini = min(a,mini)
            min_count += 1
    return [max_count,min_count]

File: breaking-the-records.java
class Result {
    public static List<Integer> breakingRecords(List<Integer> scores) {
        int max = scores.get(0), min = scores.get(0), max_count = 0, min_count = 0;
        for(int i=1;i<scores.size();i++){
            int a = scores.get(i);
            if(a>max){
                max = a;
                max_count++;
            }
            else if(a<min){
                min = a;
                min_count++;
            }
        }
        return List.of(max_count, min_count);
    }
}